corehq/apps/reports/views.py

```diff
@@ -1937,20 +1937,26 @@ def archive_form(request, domain, instance_id):
 
 
 def _get_cases_with_forms_message(domain, cases_with_other_forms, case_id_from_request):
-    def _get_case_link(case_id, name):
-        if case_id == case_id_from_request:
-            return _("%(case_name)s (this case)") % {'case_name': name}
-        else:
-            return '<a href="{}#!history">{}</a>'.format(reverse('case_data', args=[domain, case_id]), name)
+    def _get_all_case_links():
+        all_case_links = []
+        for case_id, case_name in cases_with_other_forms.items():
+            if case_id == case_id_from_request:
+                all_case_links.append(_(
+                    format_html(f'{case_name}s (this case)')
+                ))
+            else:
+                all_case_links.append(_(
+                    format_html(
+                        f'<a href=\"{reverse("case_data", args=[domain, case_id])}#!history\">{case_name}</a>'
+                    )
+                ))
+        return all_case_links
+
+    case_links = ", ".join(_get_all_case_links())
```

Comment:
This is why `format_html_join` exists. When you join `format_html` fragments like this, the string is no longer considered 'safe'. Thus, if you try to use `format_html` at the bottom, it will escape all the HTML we previously tried to preserve. We can either use a trivial `format_html_join` here, i.e.: `case_links = format_html_join(', ', '{}', _get_all_case_links())`, or we can use the join as is, with `mark_safe`: `case_links = mark_safe(', '.join(_get_all_case_links())) # nosec: links have already been sanitized`.  My preference would be `format_html_join`, as I'm mostly trying to eliminate calling `mark_safe` when possible